File: C#/0746. Min Cost Climbing Stairs.cs
public class Solution {
    public int MinCostClimbingStairs(int[] cost) {
        int[] dp=new int[cost.Length+1];
        dp[0]=0;
        dp[1]=cost[0];
        for(int i=1;i<cost.Length;i++){
            dp[i+1]=cost[i]+Math.Min(dp[i],dp[i-1]);
        }
        return Math.Min(dp[dp.Length-1],dp[dp.Length-2]);
    }
}

File: 20190519python/0746. Min Cost Climbing Stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        if len(cost)==2:
            return min(cost)
        if len(cost)<2:
            return 0
        
        dp=[]
        dp.append(cost[0])
        dp.append(cost[1])
        
        for i in range(2,len(cost)):
            dp.append(cost[i]+min(dp[-1],dp[-2]))
        return min(dp[-1],dp[-2])
